akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java

```diff
@@ -644,6 +644,40 @@ public void mustBeAbleToCombine() throws Exception {
   }
 
   @Test
+  public void mustBeAbleToZipN() throws Exception {
+    final JavaTestKit probe = new JavaTestKit(system);
+    final Source<Integer, NotUsed> source1 = Source.from(Arrays.asList(0, 1));
+    final Source<Integer, NotUsed> source2 = Source.from(Arrays.asList(2, 3));
+
+    final List<Source<Integer, ?>> sources = Arrays.asList(source1, source2);
+
+    final Source<List<Integer>, ?> source = Source.zipN(sources);
+
+    final CompletionStage<Done> future = source.runWith(Sink.foreach(elem -> probe.getRef().tell(elem, ActorRef.noSender())), materializer);
+
+    probe.expectMsgAllOf(Arrays.asList(0, 2), Arrays.asList(1, 3));
+
+    future.toCompletableFuture().get(200, TimeUnit.MILLISECONDS);
```

Comment:
Please cleanup whatever short timeouts you see.
One exception is such as expectNoMsg, those should be kept short to avoid slowing down test execution.